src/core/lib/event_engine/poller.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GRPC_CORE_LIB_EVENT_ENGINE_POLLER_H
+#define GRPC_CORE_LIB_EVENT_ENGINE_POLLER_H
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/container/inlined_vector.h"
+#include "absl/status/statusor.h"
+
+#include <grpc/event_engine/event_engine.h>
+
+namespace grpc_event_engine {
+namespace experimental {
+
+namespace {
+constexpr size_t kInitialSize = 5;
```

Comment:
move this into `class Poller` - generally we don't want anonymous namespaces in headers anyway

Reply:
Done. I left a note about maybe needing to tune this eventually, 5 is arbitrary. For IOCP, I expect a vector of size 1.